core/templates/domain/collection/collection.model.ts

```diff
@@ -32,35 +32,40 @@ interface ExplorationIdToNodeIndexMap {
 }
 
 export interface CollectionBackendDict {
-  'id': string;
-  'title': string;
-  'objective': string;
-  'language_code': string;
-  'tags': string[];
-  'schema_version': number;
+  // When creating a new collection, properties below are always
+  // initialized with null values. These are null until populated
+  // from the backend and provided themselves by the user.
+  'id': string | null;
+  'title': string | null;
+  'objective': string | null;
+  'language_code': string | null;
+  'tags': string[] | null;
+  'schema_version': number | null;
+  'category': string | null;
+  'version': number | null;
   'playthrough_dict': CollectionPlaythroughBackendDict;
```

Comment:
This is backend dict right? So technically it shouldn't have null values as we don't initialise backend dicts. We get them from the server.

Reply:
@srijanreddy98 when an empty collection is created these values are null initially 
![image](https://user-images.githubusercontent.com/60519303/122430306-1b5a5d00-cfb1-11eb-9f9f-8c7e74e46bcf.png)
The id and version need to be null but those shouldn't be empty strings as well ? We're passing ids before creating a collection see this https://github.com/oppia/oppia/blob/b59fc11db163f039d8ee7ef6f7d0184ca5856a95/core/templates/components/entity-creation-services/collection-creation.service.ts#L69

Reply:
I get this... But do we allow null to go as a backend_dict to the server? I can see red line that imply that they are required. So, we shouldn't allow such class instances to be converted to backend dicts. Title really looks necessary but for a full list of what values are accepted, you can foray into the backend code and figure it out.

Reply:
@srijanreddy98 I have confirmed this with @vojtechjelinek, we allow `null` to go as a backend dict to server.
![image](https://user-images.githubusercontent.com/60519303/122652231-5a2c1680-d15b-11eb-9e5b-d701271b000b.png)
The ID as you can see is autogenerated as soon as a user creates collection, schema_version is also assigned from the backend, we can only make `language_code` to be not null and make it `en` by default.Rest of the properties I think should have ` | null` as a type. Does this sounds right ?

Reply:
Thanks for explaining.